ui/apps/platform/src/Containers/Policies/policies.utils.ts

```diff
@@ -629,34 +629,39 @@ export function getServerPolicy(policyUntrimmed: ClientPolicy): Policy {
     return serverPolicy;
 }
 
+export type ValidPolicyLifeCycle = ['BUILD'] | ['DEPLOY'] | ['BUILD', 'DEPLOY'] | ['RUNTIME'];
+
 export function getLifeCyclesUpdates<
     T extends Pick<
         ClientPolicy,
         'lifecycleStages' | 'eventSource' | 'excludedImageNames' | 'enforcementActions'
     >,
->(values: T, lifecycleStage: LifecycleStage, isChecked: boolean): T {
+>(values: T, selectedStages: ValidPolicyLifeCycle): T {
     /*
      * Set all changed values at once, because separate setFieldValue calls
      * for lifecycleStages and eventSource cause inconsistent incorrect validation.
      */
     const changedValues = cloneDeep(values);
-    if (isChecked) {
-        changedValues.lifecycleStages = [...values.lifecycleStages, lifecycleStage];
-    } else {
-        changedValues.lifecycleStages = values.lifecycleStages.filter(
-            (stage) => stage !== lifecycleStage
-        );
-        if (lifecycleStage === 'RUNTIME') {
-            changedValues.eventSource = 'NOT_APPLICABLE';
-        }
-        if (lifecycleStage === 'BUILD') {
-            changedValues.excludedImageNames = [];
-        }
+
+    if (!selectedStages.some((stage) => stage === 'RUNTIME')) {
+        changedValues.eventSource = 'NOT_APPLICABLE';
```

Comment:
Seeing this business logic confirms need for UI team representation to participate in discussions about potential new `eventSource` which would be applicable for `'RUNTIME'` only.

Reply:
A though for which I am unsure about pro and con of type predicate helper functions:
* make explicit to reader what TypeScript infers via return type of `stages is ['RUNTIME']` and so on
* maybe reduce repetition?

Reply:
I had considered this too, I think it is worthwhile to add. It sets us up in case we need to repeat the same logic elsewhere as well.

Reply:
Lo and behold, switching to this logic revealed incidentally that one of the tests was incorrect and missed bad behavior. 🎉